### DISCUSSION TITLE
Add custom Tracking / analytics

### DISCUSSION THREAD

**Discussion by rafipiccolo** [coder/code-server#6136] [General] [answered]
How would you track a users activity ?

Like checking that the interface is open or not.
Browser Tab open or hidden.
mouse click / key press.

I searched and only saw heartbeat. But nothing I could use.

Thanks

**Comment by code-asher** [comment, marked as answer]
You might be able to get some of this information out of the telemetry.  You can send telemetry to your own server by setting the `CS_TELEMETRY_URL` environment variable.
